### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by jbaublitz**

### CODE REVIEW

**Comment by jbaublitz** [comment]
@mulkieran The most recent failure only failed on an allowed failure.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  pull_request:
+    branches:
+      - master
+      - develop-2.2.1
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  # MANDATORY CHECKS USING CURRENT DEVELOPMENT COMPILER
```
Comment (mulkieran) [inline .github/workflows/main.yml:19]: This matrix combines both checks w/ CURRENT DEVELOPMENT and checks with STABLE. Can you move "CURRENT DEVELOPMENT" comment down to line 24 and place old "STABLE" comment at line 30?

**Review by mulkieran — CHANGES_REQUESTED** [review]
Aside from the comment request it all looks good.

**Comment by jbaublitz** [comment]
Ok, this should be ready!

**Review by mulkieran — CHANGES_REQUESTED** [review]
Ok! Please close and open a final PR whenever you're ready.

**Pull request closed without merging**